Automattic/woocommerce-connect-client
# Add global notices

This PR seeks to replace the existing inline save/error notices and add global notices instead. It re-uses the Calypso `GlobalNotices` component and the associated state tree (reducer/actions). In order to get that to work properly, this PR also introduces some changes to the i18n boot/mixin system so that it works with `this.translate` in Calypso components properly.

Success:

![image](https://cloud.githubusercontent.com/assets/260253/15450331/26433e72-1f4d-11e6-8337-7a7710e0ff3f.png)

Error:

![image](https://cloud.githubusercontent.com/assets/260253/15450350/a42b0860-1f4d-11e6-92ad-8395dee1b102.png)

In addition, this PR upgrades Calypso and all its sub-dependencies to their latest versions, by upgrading our `npm-shrinkwrap.json`


To test:

* Option one is to just run `npm install`, in theory npm should do a good job of just upgrading what it needs to. Option two is to delete the `node_modules` folder and `npm cache clean && npm install`.
* Checkout this PR
* Ensure everything compiles as expected with `npm start` and `npm run dist`
* Ensure tests pass for `npm test`
* Save the USPS form with different conditions generating error and success states
* Ensure that notices are generated as expected
* Success notices should disappear after 2250ms
* Error notices persist for 7s. I chose this semi-arbitrarily, so definitely welcome to change this up.

cc @kellychoffman for a design review (there's definitely a few small css tweaks to be made here)

cc @nabsul @allendav @jeffstieler 

 Fixes #322
